.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+**Description:** Describe in a couple of sentence what this PR adds
+
+**JIRA:** Link to JIRA ticket
+
+**Reviewers:**
+- [ ] tag reviewer 
+- [ ] tag reviewer 
+
+**Merge checklist:**
+- [ ] All reviewers approved
+- [ ] CI build is green
+- [ ] Commits are squashed
+- [ ] PR author is listed in AUTHORS
+
+**Post merge:**
+- [ ] Delete working branch (if not needed anymore)
```

Comment:
Should we add a step to verify that doc changes made it to Read the Docs?

Reply:
I'm going to slightly change your proposal to include QA.
```
- [ ] Test that changes published to Read the Docs appear as expected.
```
Hopefully this will reduce friction for typo and other extremely minor fixes where people don't feel the need to check.